Serialiser_Engine/Compute/Serialise/IObject.cs

```diff
@@ -0,0 +1,61 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using MongoDB.Bson.IO;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+
+namespace BH.Engine.Serialiser
+{
+    public static partial class Compute
+    {
+
+        /*******************************************/
+        /**** Public Methods                    ****/
+        /*******************************************/
+        public static void Serialise(this IObject value, BsonDocumentWriter writer)
+        {
+            if (value == null)
+            {
+                writer.WriteNull();
+                return;
+            }
+
+            writer.WriteStartDocument();
+
+            writer.WriteName("_t");
+            writer.WriteString(value.GetType().FullName);
+
+            foreach (PropertyInfo prop in value.GetType().GetProperties())
+            {
+                writer.WriteName(prop.Name);
+                ISerialise(prop.GetValue(value), writer);
+            }
+            writer.WriteEndDocument();
```

Comment:
Would be a change to behaviour, but might be good to add a `writer.AddVersion()` here?

As I think we have discussed before, good to have access to that on all levels when it comes to versioning.

That would ofc mean that you might have issues de-serialising files using an older version of the BHoM, that have been serialised with this (not, forward compatible) so I guess could be something we could but a pin in and add in 6.3

Reply:
For the moment, I am focusing on producing the exact same content than the current version. Makes it easier to debug and catch problems. I would recommend doing anything that alters the output json in a later PR to keep things simple.